[RHDHBUGS-2722]: Restructure authentication title to JTBD outline (#2049)
redhat-developer/red-hat-developers-documentation-rhdh: modules/shared/proc-enable-authentication-with-microsoft-azure.adoc

:_mod-docs-content-type: PROCEDURE

[id="enable-authentication-with-microsoft-azure_{context}"]
= Enable authentication with {azure-brand-name}

[role="_abstract"]
Configure {azure-brand-name} as your {product} sign-in provider.

.Prerequisites
* You have xref:share-a-secret-with-microsoft-azure_{secrets-context}[shared a secret with {azure-brand-name}].

.Procedure

> . Enable {azure-short} authentication, by adding the Microsoft authentication provider to your `{my-app-config-file}` file content:

Reviewer: ```suggestion
. Add the Microsoft authentication provider to your `{my-app-config-file}` file:
```

----

`domainHint`::
. Leave this parameter empty, or enter the tenant ID when your application registration is single-tenant.
. Leave this parameter empty when your application registration is multitenant.

> . Enter the tenant ID to reduce login friction for users with accounts in multiple tenants, by automatically filtering out accounts from other tenants.

Reviewer: I'd keep this list unordered

Author: ```suggestion
* Leave this parameter empty, or enter the tenant ID when your application registration is single-tenant.
* Leave this parameter empty when your application registration is multitenant.
* Enter the tenant ID to reduce login friction for users with accounts in multiple tenants, by automatically filtering out accounts from other tenants.
```

+
For more information, see link:https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/home-realm-discovery-policy[Home Realm Discovery].

`additionalScopes`::
Enter the list of additional scopes to add scopes for the application registration.
The default and mandatory value lists following scopes:
. `openid`
. `offline_access`
. `profile`
. `email`

> . `User.Read`

Reviewer: I'd keep this list unordered

Author: ```suggestion
* `openid`
* `offline_access`
* `profile`
* `email`
* `User.Read`
```

====
In production mode, do not enable `dangerouslyAllowSignInWithoutUserInCatalog`.
====

> . To disable the guest login option, in the `{my-app-config-file}` file, set the authentication environment to `production`:

Reviewer: is this optional?

Author: See above

+
[source,yaml]
----
auth:
  environment: production
----

.Verification
. To verify {azure-short} user authentication:
.. Go to the {product-short} login page.
.. Your {product-short} sign-in page displays *Sign in using Microsoft* and the Guest user sign-in is disabled.

> .. Log in with an {azure-short} account.

Reviewer: ```suggestion
. Go to the {product-short} login page.
. Your {product-short} sign-in page displays *Sign in using Microsoft* and the Guest user sign-in is disabled.
. Log in with an {azure-short} account.
```
